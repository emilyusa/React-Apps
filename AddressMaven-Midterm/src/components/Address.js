import React, { Component } from 'react';
import '../css/App.css';
import tempAddressList from '../address-list';
import AddressShow from './AddressShow';

class Address extends Component {

    constructor(props) {
        super(props);
        this.addressList = null;
        this.addressIndex = 0;
        this.state = {
            address: tempAddressList[this.addressIndex]
        };
        this.getAddress();
    }

    getAddress=()=> {
        fetch('/address-list')
            .then((response) => response.json())
            .then((addressListFromServer) => {
                this.addressList=addressListFromServer;
            })
            .catch((ex) => {
                console.log(ex);
            })
    }

    setAddress=(offset)=>{
        this.addressIndex += offset;
        if(this.addressIndex>this.addressList.result.length-1) {
            this.addressIndex = 0;
        }
        else if(this.addressIndex<0) {
            this.addressIndex = this.addressList.result.length - 1;
        }
        this.setState({
            address:this.addressList.result[this.addressIndex]
        });
    };

    render() {
        if (!this.quiet){console.log("ADDRESS RENDER");}
        return (
            <div>
                <AddressShow
                    address={this.state.address}
                    setAddress={this.setAddress}
                />
            </div>

        );
    }
}

export default Address;