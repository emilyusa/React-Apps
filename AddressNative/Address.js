import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import styles from './elf-styles';
import AddressShow from './AddressShow';
import tempAddressList from './address-list';

export default class Address extends React.Component {

		constructor() {
			super();
			this.addressList=[{}];
			this.addressIndex=0;
			this.state = {
				address: tempAddressList[0]
			};
			this.getAddress();
		}


		getAddress=()=> {

			fetch('http://10.12.13.28:30026/address-list')
				.then((response) => response.json())
				.then((addressListFromServer) => {
					this.addressList=addressListFromServer;
                    console.log(this.addressList.result);
				})
				.catch((ex) => {
					console.log(ex);
				});
			}



		setAddress=(offset)=> {

			this.addressIndex += offset;
			if(this.addressIndex>99) {

				this.addressIndex = 0;
			}
			else if(this.addressIndex<0) {
				this.addressIndex = 99;
			}
			this.setState({
				address:this.addressList.result[this.addressIndex]
			});
		};

		render() {
			return (
				<View style={styles.container}>
					<AddressShow
						address={this.state.address}
						setAddress={this.setAddress}
					/>
				</View>
        );
    }
}

