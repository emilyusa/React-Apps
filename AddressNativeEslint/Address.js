import React from 'react';
import { View } from 'react-native';
import styles from './elf-styles';
import AddressShow from './AddressShow';
import tempAddressList from './address-list';

export default class Address extends React.Component {
    constructor(props) {
        super(props);
        this.canceled = false;
        this.state = {
            addressList: null,
            addressIndex: 0,
            address: tempAddressList[0]
        };
    }

    componentDidMount() {
        this.getAddress();
    }

    componentWillUnmount() {
        this.canceled = true;
    }

    getAddress = () => {
        fetch('http://18.188.225.33:30026/address-list')
            .then(response => response.json())
            .then(addressListFromServer => {
                if (!this.canceled) {
                    this.setState({ addressList: addressListFromServer });
                    this.setState({ index: 0 });
                    //console.log(this.addressList.result);
                }
            })
            .catch(ex => {
                console.log(ex);
            });
    };

    setAddress = (offset, goToFirst) => {
        // if (offset === 0) {
        //     const middlebutton = tempAddressList[1];
        //     this.setState({
        //         address: middlebutton
        //     });
        // } else {
        var tempIndex = this.state.addressIndex;
        if (goToFirst === 0) {
            tempIndex += offset;
            if (tempIndex > 99) {
                tempIndex = 99;
            } else if (tempIndex < 0) {
                tempIndex = 0;
            }
        } else if (goToFirst == -1) {
            tempIndex = 0;
        } else if (goToFirst == 1) {
            tempIndex = 99;
        }
        this.setState({
            addressIndex: tempIndex,
            address: this.state.addressList.result[tempIndex]
        });
        // }
    };

    render() {
        return (
            <View style={styles.container}>
                <AddressShow
                    address={this.state.address}
                    setAddress={this.setAddress}
                />
            </View>
        );
    }
}
