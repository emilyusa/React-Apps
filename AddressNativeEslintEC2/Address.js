import React from 'react';
import { View } from 'react-native';
import styles from './elf-styles';
import AddressShow from './AddressShow';
import tempAddressList from './address-list';

export default class Address extends React.Component {
    constructor(props) {
        super(props);
        this.canceled = false;
        this.state = {
            addressList: null,
            addressIndex: 0,
            address: tempAddressList[0]
        };
    }

    componentDidMount() {
        this.getAddress();
    }

    componentWillUnmount() {
        this.canceled = true;
    }

    getAddress = () => {
        fetch('http://18.188.225.33:30026/address-list')
            .then(response => response.json())
            .then(addressListFromServer => {
                if (!this.canceled) {
                    this.setState({ addressList: addressListFromServer });
                    this.setState({ index: 0 });
                }
            })
            .catch(ex => {
                console.log(ex);
            });
    };

    setAddress = offset => {
        if (offset === 0) {
            const middlebutton = tempAddressList[1];
            this.setState({
                address: middlebutton
            });
        } else {
            var tempIndex = this.state.addressIndex;
            tempIndex += offset;
            if (tempIndex > 99) {
                tempIndex = 0;
            } else if (tempIndex < 0) {
                tempIndex = 99;
            }
            this.setState({
                addressIndex: tempIndex,
                address: this.state.addressList.result[tempIndex]
            });
        }
    };

    render() {
        return (
            <View style={styles.container}>
                <AddressShow
                    address={this.state.address}
                    setAddress={this.setAddress}
                />
            </View>
        );
    }
}
