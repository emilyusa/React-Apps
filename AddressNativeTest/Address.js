import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import styles from './elf-styles';
import AddressShow from './AddressShow';
import tempAddressList from './address-list';

export default class Address extends React.Component {

    constructor(props) {
        super(props);
        this.canceled = false;
        this.state = {
            addressList: null,
            addressIndex: 0,
            address: tempAddressList[0]
        };
    }

    componentDidMount() {
        this.getAddress();
    }

    componentWillUnmount() {
        this.canceled = true;
    }

    getAddress = () => {
        fetch('http://10.0.0.18:30026/address-list')
            .then(response => response.json())
            .then(addressListFromServer => {
                if (!this.canceled) {
                    this.setState({ addressList: addressListFromServer });
                    this.setState({ index: 0 });
                }
            })
            .catch(ex => {
                console.log(ex);
            });
    };

    setAddress=(offset)=>{
        if (offset === 0) {
            const middlebutton = tempAddressList[1];
            this.setState({
                address: middlebutton
            });
        } else {
            var tempIndex = this.state.addressIndex;
            tempIndex += offset;
            if (tempIndex > 99) {
                tempIndex = 0;
            } else if (tempIndex < 0) {
                tempIndex = 99;
            }
            this.setState({
                addressIndex: tempIndex,
                address: this.state.addressList.result[tempIndex]
            });
        }
    };


    // getAddress=()=> {
    //
		// 	fetch('http://10.12.13.28:30026/address-list')
		// 		.then((response) => response.json())
		// 		.then((addressListFromServer) => {
		// 			this.addressList=addressListFromServer;
		// 		})
		// 		.catch((ex) => {
		// 			console.log(ex);
		// 		});
		// 	}
    //
    //
    //
		// setAddress=(offset)=> {
    //         if(offset==0){
    //             const middlebutton=tempAddressList[1];
    //             this.setState({
    //                 address:middlebutton
    //             });
    //         }else {
    //             this.addressIndex += offset;
    //             if (this.addressIndex > 99) {
    //
    //                 this.addressIndex = 0;
    //             }
    //             else if (this.addressIndex < 0) {
    //                 this.addressIndex = 99;
    //             }
    //             this.setState({
    //                 address: this.addressList.result[this.addressIndex]
    //             });
    //         }
		// };


		render() {
			return (
				<View style={styles.container}>
					<AddressShow
						address={this.state.address}
						setAddress={this.setAddress}
					/>
				</View>
        );
    }
}

