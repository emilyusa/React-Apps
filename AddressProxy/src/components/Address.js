import React, { Component } from 'react';
import '../css/App.css';
import tempAddressList from '../address-list';
import AddressShow from './AddressShow';

class Address extends Component {


    constructor(props) {
        super(props);
        this.debug = false;
        this.addressIndex=0;
        this.addressList = null;
        this.state = {
            address: tempAddressList[this.addressIndex]
        };
        this.getAddress();
        
    }

    getAddress=()=> {

        fetch('/address-list')
            .then((response) => response.json())
            .then((addressFromServer) => {
                console.log(addressFromServer);
                this.addressList=addressFromServer;

            })
            .catch((ex) => {
                console.log(ex);
            })
    }


    setAddress=(offset)=>{
        this.addressIndex+=offset;
        
        console.log('click showAddrees button');
        this.setState({
            address:this.addressList[this.addressIndex]
        })
    };

    render() {
        if (!this.quiet){console.log("ADDRESS RENDER");}
        return (
            <div>
                <AddressShow
                    address={this.state.address}
                    onAddressChange={this.setAddress}
                />
            </div>

        );
    }
}

export default Address;
