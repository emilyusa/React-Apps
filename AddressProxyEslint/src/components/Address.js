import React, { Component } from 'react';
import './App.css';
import tempAddressList from '../address-list';
import AddressShow from './AddressShow';

class Address extends Component {
    constructor(props) {
        super(props);
        this.canceled = false;
        this.state = {
            addressList: null,
            addressIndex: 0,
            address: tempAddressList[0]
        };
    }

    componentDidMount() {
        this.getAddress();
    }

    componentWillUnmount() {
        this.canceled = true;
    }

    getAddress = () => {
        fetch('/address-list')
            .then(response => response.json())
            .then(addressListFromServer => {
                if (!this.canceled) {
                    this.setState({ addressList: addressListFromServer });
                    this.setState({ index: 0 });
                }
            })
            .catch(ex => {
                console.log(ex);
            });
    };

    setAddress = offset => {
        if (offset === 0) {
            const middlebutton = tempAddressList[1];
            this.setState({
                address: middlebutton
            });
        } else {
            var tempIndex = this.state.addressIndex;
            tempIndex += offset;
            if (tempIndex > 99) {
                tempIndex = 0;
            } else if (tempIndex < 0) {
                tempIndex = 99;
            }
            this.setState({
                addressIndex: tempIndex,
                address: this.state.addressList.result[tempIndex]
            });
        }
    };

    render() {
        return (
            <div>
                <AddressShow
                    address={this.state.address}
                    setAddress={this.setAddress}
                />
            </div>
        );
    }
}

export default Address;
