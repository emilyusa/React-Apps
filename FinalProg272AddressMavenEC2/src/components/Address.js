import React, { Component } from 'react';
import '../App.css';
import AddressShow from './AddressShow';
import PropTypes from 'prop-types';
import PouchDB from 'pouchdb';
import addresses from '../address-list';

class Address extends Component {
    constructor(props) {
        super(props);
        this.canceled = false;
        this.state = {
            editOpen: false,
            namesIndex: 0,
            names: [
                {
                    _id: 'unknown',
                    firstName: 'unknown',
                    lastName: 'unknown',
                    street: 'unknown',
                    city: 'unknown',
                    state: 'unknown',
                    zip: 'unknown',
                    phone: 'unknown',
                    website: 'unknown',
                    email: 'unknown',
                    contact: 'unknown'
                }
            ]
        };
    }

    componentDidMount() {
        this.db = new PouchDB('address-list-chunyan');
        this.remoteCouch =
            'http://IP_Address:5984/addressLinks to an external site.';
        this.syncDom = document.getElementById('sync-wrapper');

        this.db
            .changes({
                since: 'now',
                live: true
            })
            .on('change', this.showAddress);
    }

    componentWillUnmount() {
        this.canceled = true;
    }

    watcher = event => {
        console.log('Watch Change', event);
        this.showAddress();
    };

    showAddress = () => {
        const that = this;
        return that.props.dataManager.db
            .find({
                selector: { lastName: { $gt: null } },
                sort: ['lastName']
            })
            .then(response => {
                console.log('RECORD COUNT:', response.docs.length);
                const names = response.docs.map(address => {
                    return {
                        _id: address._id,
                        _rev: address._rev,
                        firstName: address.firstName,
                        lastName: address.lastName,
                        street: address.street,
                        city: address.city,
                        state: address.state,
                        zip: address.zip,
                        phone: address.phone,
                        website: address.website,
                        email: address.email,
                        contact: address.contact
                    };
                });
                if (!this.canceled) {
                    that.setState({ names: names });
                }
            });
    };

    setAddress = offset => {
        if (offset === 0) {
            this.setState({ names: addresses[1], open: this.state.editOpen });
        } else {
            const value = this.state.namesIndex + offset;
            if (value >= 0 && value <= this.state.names.length - 1) {
                this.setState({ namesIndex: value, open: this.state.editOpen });
            }
        }
    };

    save = name => {
        console.log(name);
        this.props.dataManager
            .save(name)
            .then(function(response) {
                console.log(response);
            })
            .catch(function(err) {
                console.log(err);
            });
    };

    delete = name => {
        this.props.dataManager
            .delete(name._id)
            .then(function(result) {
                console.log('BAR', result);
            })
            .catch(function(err) {
                console.log(err);
            });
    };

    setEdits = (name, event) => {
        var data = this.props.name;
        data[name] = event.target.value;
        this.setState({ edits: data });
    };

    addressEdit = address => {
        console.log(address);

        if (!address) {
            return this.setState({ editOpen: false });
        }

        this.setState({
            edits: address,
            editOpen: false
        });
    };

    render() {
        return (
            <AddressShow
                name={this.state.names[this.state.namesIndex]}
                showAddress={this.showAddress}
                setAddress={this.setAddress}
                save={this.save}
                delete={this.delete}
            />
        );
    }
}

Address.propTypes = {
    dataManager: PropTypes.shape({
        watchChanges: PropTypes.func,
        save: PropTypes.func,
        delete: PropTypes.func
    }),
    name: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string,
        street: PropTypes.string,
        city: PropTypes.string,
        state: PropTypes.string,
        zip: PropTypes.string,
        phone: PropTypes.string,
        website: PropTypes.string,
        email: PropTypes.string,
        contact: PropTypes.string
    }),
    open: PropTypes.bool
};

export default Address;
