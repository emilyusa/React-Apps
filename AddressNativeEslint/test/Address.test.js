import React from 'react';
import Address from '../Address';
import addresses from '../address-list';
import {configure,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({adapter:new Adapter()});


describe ('Address tests',function() {

    it('renders without crashing', () => {
        const tree = renderer.create(<Address />).toJSON();
        expect(tree).toMatchSnapshot();

    });

    beforeEach(() => {
        global.fetch = jest.fn().mockImplementation(() => {
            const promise = new Promise(resolve => {
                resolve({
                    ok: true,
                    json: function () {
                        return [
                            {
                                firstName: 'Chunyan',
                                lastName: 'Li',
                                address: '4190 129th PL SE',
                                city: 'Bellevue',
                                state: 'WA',
                                zip: '98006',
                                phone: '425-247-5300',
                                website: 'www.google.com',
                                email: 'emilyusa@yahoo.com',
                                contact:'family'
                            }
                        ];
                    }
                });
            });
            return promise;
        });
    });


    const afterClickFieldTest = (wrapper, finder) => {
        setImmediate(() => {
            wrapper.update();
            wrapper.instance().setAddress(0);
            setImmediate(() => {
                wrapper.update();
                try {
                    finder();
                } catch (e) {
                    console.log(e);
                }
            });
        });
    };

    it('renders state of firstName before button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        expect(wrapper.find('AddressShow').prop('address').firstName).toEqual('unknown');
    });

    it('renders state of lastName before button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        expect(wrapper.find('AddressShow').prop('address').lastName).toEqual('unknown');
    });

    it('renders state of street before button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        expect(wrapper.find('AddressShow').prop('address').street).toEqual('unknown');
    });

    it('renders state of city before button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        expect(wrapper.find('AddressShow').prop('address').city).toEqual('unknown');
    });

    it('renders state of state before button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        expect(wrapper.find('AddressShow').prop('address').state).toEqual('unknown');
    });

    it('renders state of zip before button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        expect(wrapper.find('AddressShow').prop('address').zip).toEqual('unknown');
    });

    it('renders state of phone before button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        expect(wrapper.find('AddressShow').prop('address').phone).toEqual('unknown');
    });

    it('renders state of website before button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        expect(wrapper.find('AddressShow').prop('address').website).toEqual('unknown');
    });

    it('renders state of email before button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        expect(wrapper.find('AddressShow').prop('address').email).toEqual('unknown');
    });

    it('renders state of contact before button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        expect(wrapper.find('AddressShow').prop('address').contact).toEqual('unknown');
    });

    it('renders state of firstName after button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        afterClickFieldTest(wrapper, () => {
            expect(wrapper.find('AddressShow').prop('address').firstName).toEqual('Chunyan');
        });
    });

    it('renders state of lastName after button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        afterClickFieldTest(wrapper, () => {
            expect(wrapper.find('AddressShow').prop('address').lastName).toEqual('Li');
        });
    });

    it('renders state of street after button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        afterClickFieldTest(wrapper, () => {
            expect(wrapper.find('AddressShow').prop('address').street).toEqual('4190 129th PL SE');
        });
    });

    it('renders state of city after button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        afterClickFieldTest(wrapper, () => {
            expect(wrapper.find('AddressShow').prop('address').city).toEqual('Bellevue');
        });
    });

    it('renders state of state after button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        afterClickFieldTest(wrapper, () => {
            expect(wrapper.find('AddressShow').prop('address').state).toEqual('WA');
        });
    });

    it('renders state of zip after button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        afterClickFieldTest(wrapper, () => {
            expect(wrapper.find('AddressShow').prop('address').zip).toEqual('98006');
        });
    });

    it('renders state of phone after button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        afterClickFieldTest(wrapper, () => {
            expect(wrapper.find('AddressShow').prop('address').phone).toEqual('425-247-5300');
        });
    });

    it('renders state of website after button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        afterClickFieldTest(wrapper, () => {
            expect(wrapper.find('AddressShow').prop('address').website).toEqual('www.google.com');
        });
    });

    it('renders state of email after button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        afterClickFieldTest(wrapper, () => {
            expect(wrapper.find('AddressShow').prop('address').email).toEqual('emilyusa@yahoo.com');
        });
    });

    it('renders state of contact after button click', () => {
        const wrapper = shallow(<Address addressList={addresses}/>);
        afterClickFieldTest(wrapper, () => {
            expect(wrapper.find('AddressShow').prop('address').contact).toEqual('family');
        });
    });
});