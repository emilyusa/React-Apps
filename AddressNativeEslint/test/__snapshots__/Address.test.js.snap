// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address tests renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "powderblue",
      "flex": 1,
      "marginBottom": 1,
      "marginTop": 24,
      "padding": 50,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "steelblue",
        "flexDirection": "column",
      }
    }
  >
    <View
      id="addressShow"
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        firstName:
        unknown
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        lastName:
        unknown
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Street:
        unknown
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        City:
        unknown
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        State:
        unknown
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Zip:
        unknown
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Phone:
        unknown
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Websit:
        unknown
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Email:
        unknown
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Contact:
        unknown
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={undefined}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "darkturquoise",
                  },
                ]
              }
            >
              &lt;&lt;
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={undefined}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "darkturquoise",
                  },
                ]
              }
            >
              &lt;
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={undefined}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "darkturquoise",
                  },
                ]
              }
            >
              &gt;
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={undefined}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "darkturquoise",
                  },
                ]
              }
            >
              &gt;&gt;
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
